src/mongo/db/storage/mmap_v1/mmap_windows.cpp

```diff
@@ -184,8 +184,8 @@ void* MemoryMappedFile::map(const char* filenameIn, unsigned long long& length)
     strncpy(filename, filenameIn, 255);
     filename[255] = 0;
     {
-        size_t len = strlen(filename);
-        for (size_t i = len - 1; i >= 0; i--) {
+        int len = strlen(filename);
```

Comment:
- Think about what the function MemoryMappedFile::map does. What would it mean if you passed it the empty string for the `filenameIn` parameter? Should that be allowed?
- If the answer is "no", then we should prohibit passing in an empty string, in which case we know that strlen will return a value >= 1, so we wouldn't need to change the loop.

In other words, what I'm asking you to do is to enforce a stronger invariant on the filenameIn parameter, such that the code here is correct as written and your change is not needed.